docs/source/spec-workload-executor.rst

```diff
@@ -58,179 +58,227 @@ After accepting the inputs, the workload executor:
    to communicate to the workload executor, and ultimately the unified test
    runner, that they should stop running operations.
 
+#. MUST initialize the following variables, which will later be used to generate
+   the ``results.json`` and ``events.json`` output files:
+
+   * ``events``: Empty array of objects.
+
+   * ``errors``: Empty array of objects.
+
+   * ``failures``: Empty array of objects.
+
+   * ``numIterations``: Integer with value -1.
+
+   * ``numSuccesses``: Integer with value -1.
+
+   Note: ``numErrors`` and ``numFailures`` are intentionally omitted here as
+   they will be derived directly from ``errors`` and ``failures``.
+
 #. MUST invoke the unified test runner to execute the workload.
    If the workload includes a ``loop`` operation, the workload will run until
    terminated by the workload executor; otherwise, the workload will terminate
    when the unified test runner finishes executing all of the operations.
    The workload executor MUST handle the case of a non-looping workload and
    it MUST terminate if the unified test runner completely executes the
    specified workload.
-   
+
    If the unified test runner raises an error while executing the workload,
    the error MUST be reported using the same format as errors handled by the
    unified test runner, as described in the unified test runner specification
-   under the ``loop`` operation. Errors handled by the workload
-   executor MUST be included in the calculated (and reported) error count.
-   
+   under the ``loop`` operation. The error MUST be appended to the ``errors``
+   array.
```

Comment:
I chose to use "append" whenever we talk about modifying the arrays to avoid any race condition in overwriting the `errors` array. That could happen if the workload executor caught a propagated error from the test runner and also fetched errors from the entity map. For instance, PHPLIB's test runner always invokes the entity map callback (used by our workload executor to fetch values) after a test, even if it ultimately throws an exception. We invoke the callback from a `finally` block.

While it's unlikely that a single test would _both_ report errors via an entity _and_ propagate an error, it's theoretically possible (e.g. `loop` that catches errors and then some error in a separate operation after the `loop`).